### ISSUE TITLE
Entries with parentheses don't show up in searches

### ISSUE THREAD

**Issue by ctschroeder** [KELLIA/dictionary#30]
Entries with headwords containing parentheses don't always come up in a search.  For example, searching for ⲟⲩⲟϭⲉ retrieves zero results, and also it has zero "related entries."  However, a search for the english translation ( ["cheek"](https://corpling.uis.georgetown.edu/coptic-dictionary/results.cgi?quick_search=cheek) ) finds the headword ⲟⲩⲟ(ⲟ)ϭⲉ.

**Comment by amir-zeldes** [comment]
Yes, the headword in the XML files seems to be:

```
          <form type="lemma" xml:id="CF12678">
            <orth>ⲟⲩⲟ(ⲟ)ϭⲉ</orth>
          </form>
```

The lemma in ANNIS seems to be `ⲟⲩⲟϭⲉ`, so there's no way for it to find this entry. I think the ideal solution is to list both forms `ⲟⲩⲟϭⲉ` and `ⲟⲩⲟⲟϭⲉ` in the lexicon as forms, and choose one of them as the lemma. What do you think @phoenix-mossimo ?

**Comment by phoenix-mossimo** [comment]
I can easily create another form (will have to add a form ID (CF...) later), but there are many cases like this. Currently  ⲟⲩⲟⲟϭⲉ will find ⲟⲩⲟ(ⲟ)ϭⲉ. Is there a way to get ⲟⲩⲟ(ⲟ)ϭⲉ also with ⲟⲩⲟϭⲉ?

If not, I guess I have to create two separate forms for all forms of the type "xx(y)xxx": 1) "xxyxxx", 2) "xxxxx"

**Comment by amir-zeldes** [comment]
I think having multiple forms is the correct way, since we do it for all other variant forms. Why should an extra 'o' be different from any other variant form? It also prevent quantitative analysis, if later we want to find out how many forms each word has, forms on average per dialect, or anything else.

**Comment by phoenix-mossimo** [comment]
OK.

**Comment by amir-zeldes** [comment]
Parentheses are now ignored in searches, and ⲟⲩⲟ(ⲟ)ϭⲉ is found with or without parentheses. However I still feel this entries and others like it should be changed as outlined above. Can this issue be closed or do we need it as a reminder for the XML version?

**Comment by phoenix-mossimo** [comment]
Leave it. It is a bit more complicated than I thought, but I will try to fix it with the next release

**Comment by somiyagawa** [comment]
Maxim> Jerusalem side (GIF project) is also making multiple entries for an
entry with parenthesis. How do you fix it with the next release?

**Comment by phoenix-mossimo** [comment]
Hi, 

I attached a table where I organized all forms having parenthesis according to the criteria, relevant for the auto-correction: 

A) Form type = lemma or Form type != lemma  
B) Parenthesis preceded by: 
	1)  not a whitespace
	2)  whitespace: 
		- not at word beginning
		- at word beginning

Basically, I want to remove parenthesis and insert two forms instead in all cases except 2.1, where from in parenthesis is a preposition / adverb . These in most cases are additionally tagged as <gram type="collocPrep">ⲉϩⲣⲁⲓ</gram>, <gram type="collocAdv">ⲉⲡⲉⲥⲏⲧ</gram>

So, in which cases does GIF project use parenthesis? 
[Brackets.pdf](https://github.com/KELLIA/dictionary/files/1681357/Brackets.pdf)

**Comment by somiyagawa** [comment]
Hi Maxim, sorry for my delay. We made the entries of all the possibilities. So, we copied the entries, duplicated them if needed, modified them so that every possible form has a row each. So, we duplicated ϭⲓ (ⲙⲡ)ⲁⲓⲕ and made rows such as (1) ϭⲓⲁⲓⲕ and (2) ϭⲓ ⲙⲡⲁⲓⲕ.

**Comment by dwerning** [comment]
Obviously, the entry type with parenteses is (regrettably) still there in the new full lexicon. 
Concernings this state, you, @amir-zeldes, once wrote:

> Parentheses are now ignored in searches, and ⲟⲩⲟ(ⲟ)ϭⲉ is found with or without parentheses. 

Indeed this is not true (anymore?). Currently, the entry "ⲟⲩⲟ(ⲟ)ϭⲉ" is found via "ⲟⲩⲟⲟϭⲉ" but not via "ⲟⲩⲟϭⲉ". Can you check?

**Comment by somiyagawa** [comment]
Hi Maxim,
> So, in which cases does GIF project use parenthesis?
sorry, I made this draft last year, but obviously, I failed to send it. We
(GIF project) made the entries of all the possibilities. So, we copied the
entries, duplicated them if needed, modified them so that every possible
form has a row each. Thus, we duplicated ϭⲓ (ⲙⲡ)ⲁⲓⲕ and made rows such as
(1) ϭⲓⲁⲓⲕ and (2) ϭⲓ ⲙⲡⲁⲓⲕ.   I am not sure how the progress on the Israeli
side of the GIF project is going. We used old data last year. Probably, we
should update according to the new rules for parenthesis.

**Comment by phoenix-mossimo** [comment]
> Obviously, the entry type with parenteses is (regrettably) still there in the new full lexicon.
> Concernings this state, you, @amir-zeldes, once wrote:
> 
> > Parentheses are now ignored in searches, and ⲟⲩⲟ(ⲟ)ϭⲉ is found with or without parentheses.
> 
> Indeed this is not true (anymore?). Currently, the entry "ⲟⲩⲟ(ⲟ)ϭⲉ" is found via "ⲟⲩⲟⲟϭⲉ" but not via "ⲟⲩⲟϭⲉ". Can you check?

I guess it means that parenthesis themselves are ignored, but not their content. Otherwise we would have a problem with compound verbs (e.g. ⲁⲗⲉ (ⲉϩⲣⲁⲓ) ⲁⲙⲁϩⲧⲉ (ⲉϩⲣⲁⲓ) ⲉϫⲛ- ⲃⲱⲕ (ⲉⲡⲉⲥⲏⲧ) ⲉⲓⲛⲉ (ⲉϫⲛ- (ⲧⲟⲟⲧ⸗, ϭⲓϫ)) or entries, attested only in the bound form, for which whose absolute form was presupposed ((ⲕⲱⲧϣ)). The use of brackets is unsystematic from the conception of the lexicon.

**Comment by amir-zeldes** [comment]
Yes, what @phoenix-mossimo describes is correct: the search ignores parentheses, not their content. I also believe that these forms should be spelled out in separate form elements, since this also prevents correct usage statistics by form and various other linking functions with corpora.

**Comment by phoenix-mossimo** [comment]
As said, I will look again if duplicating the the forms can be done automatically (which is not a trivial task considering the number of parameters involved, see above https://github.com/KELLIA/dictionary/issues/30#issuecomment-361894406 ). If not - it is a painstaking manual copy-paste work, which can also be done by HiWi / Werkstudent capable of Coptic.

**Comment by dwerning** [comment]
Duplicating entries implies a thorough treatment of the _stable_ IDs. I believe we have to delay this until milestone 2.1.

**Comment by amir-zeldes** [comment]
I agree - I would love to see this fixed, but I think we shouldn't delay the 2.0.0 release for this, as it involves substantial work and could take a while. I plan to resolve #1 and propose the release after that.

**Comment by phoenix-mossimo** [comment]
I created a new kanban board for this task (https://github.com/KELLIA/dictionary/projects/3 . 
The task description of the involved parameters are summarized on Wiki pages: https://github.com/KELLIA/dictionary/wiki/Project:-Remove-brackets-(parenthesis)-from-orthographic-variants

**Comment by phoenix-mossimo** [comment]
- Parenthesis-problem resolved according to the description in  https://github.com/KELLIA/dictionary/wiki/A0:-Project:-Remove-brackets-from-orthographic-variants. 

- 799 new forms were created (will be assigned IDs). 

- The remaining forms , containing parenthesis, all denote collocated prepositions / adverbs: https://github.com/KELLIA/dictionary/wiki/A3:-The-remaining-forms-containing-brackets., 

- This is in line with TLA notation: e.g. http://aaew.bbaw.de/tla/servlet/GetWcnDetails?u=guest&f=0&l=0&wn=855910&db=0. Will be released with CCL V 1.1 in February/March.
